src/lib/mina_lib/mina_lib.ml

```diff
@@ -1179,6 +1198,26 @@ let create ?wallets (config : Config.t) =
                     ; kill_ivar= Ivar.create () }
                   , config.snark_work_fee ) )
           in
+          let%bind uptime_snark_worker_opt =
+            (* if uptime URL provided, run uptime service SNARK worker *)
+            Option.value_map config.uptime_url ~default:(return None)
+              ~f:(fun _url ->
+                Monitor.try_with ~here:[%here]
+                  ~rest:
+                    (`Call
+                      (fun exn ->
+                        let err = Error.of_exn ~backtrace:`Get exn in
```

Comment:
Does it make sense to crash here? It seems like that (or restarting the worker) might be preferable to having it die and seeing your uptime fall off a cliff, despite still being up.